### PR TITLE
implement __eq__ for LocalStore

### PR DESCRIPTION
**Pull request by Charoula-Kyriakides**
implement __eq__ for LocalStore

### CODE REVIEW

File: src/zarr/v3/store/local.py [hunk 1]
```diff
@@ -67,6 +67,9 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"LocalStore({repr(str(self))})"
 
+    def __eq__(self, other: 'LocalStore') -> bool:
```
Comment (d-v-b) [inline src/zarr/v3/store/local.py:70]: After adding `from typing import Self` to the top of the file, I think you can do this:

```suggestion
    def __eq__(self, other: Self) -> bool:
```

**Comment by d-v-b** [comment]
@Charoula-Kyriakides thanks for the contribution! It might be good to have some more information in the title and description of the PR. I can fill those in if you like.

**Comment by Charoula-Kyriakides** [comment]
That'd be great, thanks. I'm still learning. Should it be something similar
to the commit message?

**Comment by d-v-b** [comment]
> That'd be great, thanks. I'm still learning. Should it be something similar to the commit message?

Maybe something simple like "implement `__eq__` for `LocalStore`"

**Review by d-v-b — APPROVED** [review]

**Pull request merged**